code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -316,6 +318,42 @@
 		icon_state = icon_opened
 
 
+/obj/structure/closet/resisted_against(datum/source, mob/living/resister)
+	container_resist(resister)
+
+
+/obj/structure/closet/proc/container_resist(mob/living/user)
+	if(opened)
+		return
+	if(!welded && !locked)
+		open()
+		return
+	if(user.action_busy) //Already resisting or doing something like it.
+		return
+	//okay, so the closet is either welded or locked... resist!!!
+	user.changeNext_move(CLICK_CD_BREAKOUT)
+	user.last_special = world.time + CLICK_CD_BREAKOUT
+	user.visible_message("<span class='warning'>[src] begins to shake violently!</span>", \
+		"<span class='notice'>You lean on the back of [src] and start pushing the door open... (this will take about [DisplayTimeText(breakout_time)].)</span>", \
+		"<span class='italics'>You hear banging from [src].</span>")
+	if(do_after(user, breakout_time, target = src))
+		if(!user || user.stat != CONSCIOUS || user.loc != src || opened || (!locked && !welded) )
```

Comment:
Isn't these checks inside the `do_after` if not i believe do_after has `extra_checks` would that not work?

Reply:
not really, extra checks is a callback

Reply:
To be fair some of these are excessive. This is /tg/ code, and not the best of it.